worker/pkg/benthos/sql/output_sql_insert.go

```diff
@@ -283,8 +292,41 @@ func (s *pooledInsertOutput) WriteBatch(ctx context.Context, batch service.Messa
 
 	query := s.buildQuery(insertQuery)
 	if _, err := s.db.ExecContext(ctx, query); err != nil {
-		return err
+		if !s.skipForeignKeyViolations || !neosync_benthos.IsForeignKeyViolationError(err.Error()) {
+			return err
+		}
+		err = s.RetryInsertRowByRow(ctx, processedCols, processedRows)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *pooledInsertOutput) RetryInsertRowByRow(
+	ctx context.Context,
+	columns []string,
+	rows [][]any,
+) error {
+	errorCount := 0
+	insertCount := 0
+	for _, row := range rows {
+		insertQuery, err := querybuilder.BuildInsertQuery(s.driver, fmt.Sprintf("%s.%s", s.schema, s.table), columns, [][]any{row}, &s.onConflictDoNothing)
```

Comment:
`fmt.Sprintf("%s.%s", s.schema, s.table)`
I know this gets fed into goqu and properly escaped, but what if the schema or table has a period in it? I'm guessing it doesnt know what to do. Anyways, idk if we are doing this anywhere but now that I'm hyper aware of special chars thought i'd point this out...

Reply:
Maybe in a follow up we should update those `BuildInsertQuery` and `BuildUpdateQuery` functions to call out schema and table as specific parameters so that we can use goqu to build them like `goqu.T(table).Schema(schema)` so that it can handle the escaping for us.